src/infrastructure/transaction/CreateTransactionFromDTO.ts

```diff
@@ -122,6 +122,23 @@ const extractMessage = (message: any): PlainMessage | EncryptedMessage => {
     return msgObj;
 };
 
+/**
+ * @internal
+ * Extract transaction meta data
+ *
+ * @param meta - Transaction meta data
+ * @param id - TransactionId
+ * @return {TransactionInfo | AggregateTransactionInfo | undefined}
+ */
+const extractTransactionMeta = (meta: any, id: string): TransactionInfo | AggregateTransactionInfo | undefined => {
+    if (!meta) {
+        return undefined;
+    }
+    if (meta.aggregateHash || meta.aggregateId) {
+        return new AggregateTransactionInfo(UInt64.fromNumericString(meta.height), meta.index, id, meta.aggregateHash, meta.aggregateId);
```

Comment:
For the ids, I would do `id || meta.id`. The id can be returned in either the root level or inside the transaction depending on the called endpoint

https://github.com/nemtech/symbol-openapi/issues/145